qiskit/algorithms/minimum_eigen_solvers/numpy_minimum_eigen_solver.py

```diff
@@ -104,20 +64,12 @@ def supports_aux_operators(cls) -> bool:
 
     def compute_minimum_eigenvalue(
             self,
-            operator: Optional[Union[OperatorBase, LegacyBaseOperator]] = None,
+            operator: Union[OperatorBase, LegacyBaseOperator],
             aux_operators: Optional[List[Optional[Union[OperatorBase,
                                                         LegacyBaseOperator]]]] = None
     ) -> MinimumEigensolverResult:
         super().compute_minimum_eigenvalue(operator, aux_operators)
-        return self._run()
-
-    def _run(self) -> MinimumEigensolverResult:
-        """
-        Run the algorithm to compute up to the minimum eigenvalue.
-        Returns:
-            dict: Dictionary of results
-        """
-        result_ces = self._ces.run()
+        result_ces = self._ces.compute_eigenvalues(operator, aux_operators)
         self._ret = self._ces._ret  # TODO remove
```

Comment:
Do you know why does it says remove here? 🤔

Reply:
no, but it makes sense :-)

Reply:
Imagine it is because _ret was usually some internal dict where the result was built up originally and later set into the result objects that were later added. The intent was to remove that in favor of more directly building the result.

Reply:
I updated this code now that self._ret is a real object and not a dictionary.